File: app/src/main/java/com/erencol/sermon/view/adapters/SermonAdapter.java
package com.erencol.sermon.view.adapters;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.erencol.sermon.R;
import com.erencol.sermon.model.Sermon;
import com.erencol.sermon.view.adapters.ViewHolders.SermonViewHolder;
import com.erencol.sermon.databinding.SermonCellBinding;
import java.util.Collections;
import java.util.List;

public class SermonAdapter extends RecyclerView.Adapter<SermonViewHolder> {
    private List<Sermon> sermonList;
    private boolean isPremium;
    private static final int FREE_SERMON_LIMIT = 5;

    public SermonAdapter() {
        this.sermonList = Collections.emptyList();
        this.isPremium = false;
    }

    public void setPremium(boolean isPremium) {
        this.isPremium = isPremium;
        notifyDataSetChanged();
    }

    @Override
    public SermonViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        SermonCellBinding sermonCellBinding =
                DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                        R.layout.sermon_cell, parent, false);
        return new SermonViewHolder(sermonCellBinding);
    }

    @Override
    public void onBindViewHolder(final SermonViewHolder holder, final int position) {
        Sermon sermon = sermonList.get(position);
        boolean isLocked = !isPremium && position >= FREE_SERMON_LIMIT;
        holder.bindSermon(sermon, isLocked, position);

    }

    @Override
    public int getItemCount() {
        if(sermonList!=null)
            return sermonList.size();
        else
            return 0;
    }
    public void setSermonList(List<Sermon> sermonList) {
        this.sermonList = sermonList;
        notifyDataSetChanged();
    }

}

File: app/src/main/java/com/erencol/sermon/view/main/SermonAdapter.kt
package com.erencol.sermon.view.main

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.erencol.sermon.R
import com.erencol.sermon.databinding.SermonCellBinding
import com.erencol.sermon.model.Sermon

class SermonAdapter : RecyclerView.Adapter<SermonViewHolder>() {
    private var sermonList: List<Sermon> = emptyList()
    var isPremium: Boolean = false
        set(value) {
            field = value
            notifyDataSetChanged()
        }
    var onPremiumContentClick: (() -> Unit)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SermonViewHolder {
        val sermonCellBinding = DataBindingUtil.inflate<SermonCellBinding>(
            LayoutInflater.from(parent.context),
            R.layout.sermon_cell, parent, false
        )
        return SermonViewHolder(sermonCellBinding)
    }

    override fun onBindViewHolder(holder: SermonViewHolder, position: Int) {
        val isLocked = !isPremium && position >= 5
        holder.bindSermon(sermonList[position], isLocked, onPremiumContentClick)
    }

    override fun getItemCount(): Int {
        return sermonList.size
    }

    fun setSermonList(sermonList: List<Sermon>?) {
        this.sermonList = sermonList ?: emptyList()
        notifyDataSetChanged()
    }
}